src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -727,10 +727,11 @@ private static int WriteHeader(MessageOpcode opcode, byte[] sendBuffer, ReadOnly
                         _receivedMaskOffsetOffset = ApplyMask(payloadBuffer.Span.Slice(0, totalBytesReceived), header.Mask, _receivedMaskOffsetOffset);
                     }
                     header.PayloadLength -= totalBytesReceived;
+                    var endOfMessage = header.Fin && header.PayloadLength == 0;
```

Comment:
I know you got feedback to cache this, but I don't think it's the right trade-off.

This `endOfMessage` "local"'s lifetime spans an await, written to here and then read later on after an await.  That means the C# compiler needs to lift this to the state machine.  While it's "only" a `bool`, IIRC the fields on the state machine already have it word aligned, such that adding this `bool` will actually increase the size of the state machine by 8 bytes on 64-bit.  That means we're paying an extra 8 bytes of memory in order to save computing a Boolean && and an Int32 ==, which is cheap, and even then we're only saving one when the Opcode is Text.

I suggest this be undone and just go back to having `header.Fin && header.PayloadLength == 0` at the `TryValidateUtf8` and `GetResult` call sites.

Reply:
I was actually wondering about that but I saw other locals so it didn’t seem like a big deal